bin/demeuk.py

```diff
@@ -166,36 +166,102 @@
 from tqdm import tqdm
 from unidecode import unidecode
 
-
-version = '4.3.0'
+# Type hinting
+HTML_ENTITIES: dict[str, str]
+
+
+# Configuration TypedDict
+class Config(TypedDict):
+    input_encoding: list[str]
+    output_encoding: str
+    cut: bool
+    delimiter: str | list[str]
+    cut_fields: str
+    verbose: bool
+    debug: bool
+    progress: bool
+    limit: bool | int
+    skip: bool | int
+    encode: bool
+    mojibake: bool
+    tab: bool
+    trim: bool
+    newline: bool
+    hex: bool
+    html: bool
+    html_named: bool
+    umlaut: bool
+    non_ascii: bool
+    title_case: bool
+    lowercase: bool
+    punctuation: str
+    check_length: bool
+    check_min_length: int
+    check_max_length: int
+    check_controlchar: bool
+    check_case: bool
+    check_email: bool
+    check_hash: bool
+    check_mac_address: bool
+    check_non_ascii: bool
+    check_replacement_character: bool
+    check_starting_with: Literal[False] | list[str]
+    check_uuid: bool
+    check_ending_with: Literal[False] | list[str]
+    check_empty_line: bool
+    check_regex: Literal[False] | list[str]
+    check_min_digits: int
+    check_max_digits: int | float
+    check_min_uppercase: int
+    check_max_uppercase: float | int
+    check_min_specials: int
+    check_max_specials: float | int
+    add_lower: bool
+    add_latin_ligatures: bool
+    add_split: bool
+    add_umlaut: bool
+    add_without_punctuation: bool
+    remove_strip_punctuation: bool
+    remove_punctuation: bool
+    remove_email: bool
+
+    googlengram: bool
+    leak: bool
+    leak_full: bool
+
+class InputFileData(TypedDict):
+    filename: str | BinaryIO
+    chunk_estimation: int
+
+version = "4.3.0"
 
 # Search from start to finish for the string $HEX[], with block of a-f0-9 with even number
 # of hex chars. The first match group is repeated.
 HEX_REGEX = re_compile(r"^\$(?:HEX|hex)\[((?:[0-9a-fA-F]{2})+)\]$")
-EMAIL_REGEX = '.{1,64}@([a-zA-Z0-9_-]{1,63}\\.){1,3}[a-zA-Z]{2,6}'
-HASH_HEX_REGEX = '^[a-fA-F0-9]+$'
-MAC_REGEX = '^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$'
-UUID_REGEX = '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
+EMAIL_REGEX = r"[a-zA-Z0-9][a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]{0,63}@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z]{2,6})+"
```

Comment:
The local part validation is stricter, only allowing valid characters in accordance with standard email formats.

Reply:
fixes #20